test/integration/package_id/compatible_test.py

```diff
@@ -400,3 +400,32 @@ def package_id(self):
 
         c.run("install --requires=pdfium/2020.9 -pr=myprofile -s build_type=Debug")
         assert "missing. Using compatible package" in c.out
+
+    def test_compatibility_erase_package_id(self):
+        c = TestClient()
+        conanfile = textwrap.dedent("""
+            from conan import ConanFile
+
+            class PdfiumConan(ConanFile):
+                name = "diligent-core"
+                version = "1.0"
+                settings = "compiler"
+
+                def package_id(self):
+                    self.info.settings.compiler.runtime = "foobar"
```

Comment:
Simplified from https://docs.conan.io/2/reference/binary_model/custom_compatibility.html#information-erasure-in-package-id-method